code/client/clrcore-v2/Interop/ReferenceFunctionManager.cs

```diff
@@ -137,18 +138,19 @@ internal unsafe static byte[] Invoke(int reference, byte* arguments, uint argsSi
 		{
 			if (s_references.TryGetValue(reference, out var funcRef))
 			{
-				var args = MsgPackDeserializer.DeserializeArray(arguments, argsSize);
+				var deserializer = new MsgPackDeserializer(arguments, argsSize, null);
 
 				object result = null;
 
 				try
 				{
 					// there's no remote invocation support through here
-					result = funcRef.m_method(default, args);
+					result = funcRef.m_method(default, ref deserializer);
 				}
 				catch (Exception ex)
 				{
-					Debug.WriteException(ex, funcRef.m_method, args, "reference function");
+					//Debug.WriteException(ex, funcRef.m_method, args, "reference function");
+					Debug.WriteLine(ex);
```

Comment:
This change can be reverted to `WriteException` again?

Reply:
yeah it should be i'll check it as soon as i get home and commit.